codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,205 @@ public TextHeaders clear() {
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.setObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.setObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        private CharSequence commaSeparated(Object... values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    doubleQuoted(singleValue, values[i]).append(", ");
+                }
+                if (values.length > 1) {
+                    doubleQuoted(singleValue, values[end]);
+                } else {
+                    unquoted(singleValue, values[end]);
+                }
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence commaSeparated(Iterable<?> values) {
+            StringBuilder singleValue = new StringBuilder();
+            Iterator<?> iterator = values.iterator();
+            if (iterator.hasNext()) {
+                Object next = iterator.next();
+                while (iterator.hasNext()) {
+                    doubleQuoted(singleValue, next).append(", ");
+                    next = iterator.next();
+                }
+                if (singleValue.length() > 0) {
+                    doubleQuoted(singleValue, next);
+                } else {
+                    unquoted(singleValue, next);
+                }
+            }
+            return singleValue.toString();
+        }
+
+        private StringBuilder doubleQuoted(StringBuilder singleValue, Object value) {
+            CharSequence converted = valueConverter.convertObject(value);
+            return converted.charAt(0) == DOUBLE_QUOTE ?
```

Comment:
We may have to do more work when escaping values into a CSV format.  For example see `quoted-string` from https://tools.ietf.org/html/rfc7230#section-3.2.6.  If we have to look the contents of each string I'm wondering if we should also conditionally escape the values (i.e only double-quote and escape if the value contains a `,` character).  WDYT?